### PR TITLE
circulation: fix display of patron in checkin view

### PR DESCRIPTION
**Pull request by AoNoOokami**
* Fixes the display of patron info (only after a checkin).
* Fixes alignment between checked-in and checked-out items.
* Fixes space after pickup location name.
* Fixes flash messages for checkin.
* Removes unused code.
* Closes rero/rero-ils#793.
* Closes rero/rero-ils#827.
* Closes rero/rero-ils#830.
* closes rero/rero-ils#829.

Co-Authored-by: Alicia Zangger <alicia.zangger@rero.ch>

## Why are you opening this PR?

To close issues mentioned above.

## How to test?

Needs https://github.com/rero/rero-ils/pull/824

### Check item in transit for pickup
* As a (system) librarian of a library A, make a checkout of an item to a patron (for example Giulia)
* As an other patron (Simonetta), make a request on this item with a pickup location different from library A.
* As (system) librarian, make a checkin of the item in the checkin form (User services > Circulation)

Expected behaviour: the item should go in transit, and the information of the first patron should be displayed (for this example, Giulia). The right destination should be displayed too.

### Check item in transit to house
* As a (system) librarian of a library A, make a checkout of an item to a patron (for example Simonetta)
* Switch to another library
* Make a checkin of the item

Expected behaviour: the item should go in transit, and the information of the first patron should be displayed  (for this example, Simonetta). The right destination should be displayed too.

### Check other actions on both views
* Perform some actions in Circulation view of a patron (put an item in transit for pickup and to house) to confirm that the destination is correctly displayed.
* In the checkin form (User services > Circulation), perform a receive by checking in an item in transit in the destination library. The patron info should not be displayed.

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: projects/admin/src/app/circulation/checkout/checkout.component.ts [hunk 1]
```diff
@@ -24,22 +24,23 @@ import { map } from 'rxjs/operators';
...
   templateUrl: './checkout.component.html'
 })
 export class CheckoutComponent implements OnInit {
-  public placeholder: string = this.translate.instant(
+  public placeholder: string = this._translate.instant(
     'Please enter a patron card number or an item barcode.'
   );
   public searchText = '';
   public patronInfo: User;
   public isLoading = false;
```
Comment (sebastiendeleze) [inline projects/admin/src/app/circulation/checkout/checkout.component.ts]: you can remove all _public_ instructions as it's the default.

File: projects/admin/src/app/circulation/checkout/checkout.component.ts [hunk 2]
```diff
@@ -50,30 +51,32 @@ export class CheckoutComponent implements OnInit {
...
   ngOnInit() {
-    this.loggedUser = this.userService.getCurrentUser();
-    this.patronService.currentPatron$.subscribe(
+    this._loggedUser = this._userService.getCurrentUser();
+    this._patronService.currentPatron$.subscribe(
       patron => (this.patronInfo = patron)
     );
     this.searchInputFocus = true;
+    this.currentLibraryPid = this._userService.getCurrentUser().getCurrentLibrary();
+    this.patronInfo = null;
```
Comment (sebastiendeleze) [inline projects/admin/src/app/circulation/checkout/checkout.component.ts]: you can initialize to null during declaration instead of here.

**Comment by AoNoOokami** [comment]
> 
> 
> The second case "Check item in transit to house" doesn't work.
> I did the test locally:
> step 1 : login with account of Elena (Avise library)
> step 2 : checkout barcode 10000000009 for patron "Alain Premand" (10000001)
> step 3 : login with account Jean (valdôtaine library)
> step 4 : checkin barcode 10000000009 => In transit (to house) OK
> step 5 : login with account of Elena (Avise library)
> step 6 : checkin barcode 10000000009 (item is back to Avise library) => problem.
> 
> Current behaviour:
> At checkin, there is no reaction in the interface and the browser console display the error
> "_TypeError: "You provided 'undefined' where a stream was expected. You can provide an Observable, Promise, Array, or Iterable._"
> 
> Expected behaviour :
> the box displaying the item "checkin" with the status "at desk"

Ok, thanks for the test. As I understand, there is an error when a receive is performed whithout the 'switch library' menu.

**Comment by AoNoOokami** [comment]
UPDATE: the issue mentioned above is due to the overwrite of the transaction_location_pid during a receive in the backend. This is due to a programming conflict between two different use cases (`receive of an item in transit to house` and `checkin (?) of an item in the wrong library`). @BadrAly is searching a solution to solve this in https://github.com/rero/rero-ils/pull/824

Ping @pronguen @iGormilhit : It confirms that the review of the state chart is needed and should help.

**Pull request merged**